File: Boyer-Moore Majority Vote Algortihm/C/Boyer-Moore-Majority-Vote-Algortihm.c
#include <stdio.h>

int find_majority_element(int arr[], int n) {
    int candidate = 0;
    int count = 0;
    
    for (int i = 0; i < n; i++) {
        if (count == 0) {
            candidate = arr[i];
        }
        count += (arr[i] == candidate) ? 1 : -1;
    }
    
    return candidate;
}

int main() {
    int arr[] = {2, 2, 1, 1, 2, 2};
    int n = sizeof(arr) / sizeof(arr[0]);
    
    int result = find_majority_element(arr, n);
    printf("Majority element is: %d\n", result);
    
    return 0;
}

File: Boyer-Moore Majority Vote Algorithm/Java/Boyer-Moore-Majority-Vote-Algortihm.java
public class MajorityElement {

    public static int findMajorityElement(int[] arr) {
        int candidate = 0;
        int count = 0;

        for (int num : arr) {
            if (count == 0) {
                candidate = num;
            }
            count += (num == candidate) ? 1 : -1;
        }

        return candidate;
    }

    public static void main(String[] args) {
        int[] arr = {2, 2, 1, 1, 2, 2};

        int result = findMajorityElement(arr);
        System.out.println("Majority element is: " + result);
    }
}
